### PR TITLE
Users & rights module

### PR DESCRIPTION
**Pull request by Grenadingue**
Right now (02/03/16), we almost have the client side of the user & rights API

What's setup ?
- database and models
- a default user group
- a default admin group
- a default admin account belonging to the default admin group
- user sign-up with automatic association to the default user group
- user login/logout with browser sessions

What's missing ?
- ~~Get/Update user profile (user side)~~
- Rights application (server & user side)
- ~~Create/Get/Update/Delete users (admin side)~~
- Create/Get/Update/Delete groups (admin side)
- Create/Get/Update/Delete rights (admin side)
- Automatic generated user password sent by email

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -13,10 +13,15 @@
     "connect-flash": "^0.1.1",
     "connect-session-sequelize": "^3.0.0",
     "cookie-parser": "^1.4.1",
+    "ejs": "^2.4.1",
     "express": "^4.12.4",
     "express-session": "^1.13.0",
     "istanbul": "^0.4.2",
     "jasmine": "^2.4.1",
+    "js-sha256": "^0.3.0",
+    "mysql": "^2.10.2",
```
Comment (Grenadingue) [inline server/package.json:22]: @Grenadingue @leroy-p 
Update `mysql` dependency to dev-dependency.

File: server/views/logged-in-logout-poc.ejs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8">
+    <title>SAM Solution | Logged-in/Logout POC</title>
```
Comment (Grenadingue) [inline server/views/logged-in-logout-poc.ejs:5]: @Grenadingue @leroy-p 
Rename all the POCs (titles and file names) by decent names for a production environment.

File: server/adapters/groups.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+var Sequelize = null;
```
Comment (BernardJeremy) [inline server/adapters/groups.js:1]: I think it's FAR better to use three require than just set three variable in the file. And then you can get rid of the init() function.

Comment (Grenadingue) [inline server/adapters/groups.js:1]: Hmm, I was very afraid of Node.js multiple file/module inclusions, but after some researches I found that you're right! Node actually performing caching on the modules it loads, with `require()` it's okay to do that. https://nodejs.org/docs/latest/api/modules.html#modules_caching

**Pull request closed without merging**